Sources/SwiftProtobuf/Enum.swift

```diff
@@ -17,7 +17,7 @@
 
 import Swift
 
-public protocol Enum: RawRepresentable, Hashable, CustomDebugStringConvertible, FieldType, MapValueType {
+public protocol Enum: RawRepresentable, Hashable, CustomDebugStringConvertible {
     init()
     init?(jsonName: String)
     init?(protoName: String)
```

Comment:
It can't be private to the library:  This is the equivalent of `ProtoNameProviding` on `Message` -- it's how the generated code exposes literal names of `Enum` cases.

It does feel like we should be able to somehow reuse `FieldNameMap` here to provide name/number conversions.